apps/designer/app/designer/shared/fonts-manager/fonts-manager.tsx

```diff
@@ -1,18 +1,146 @@
+  // We can have 2+ assets with the same family name, so we use a map to dedupe.
+  const uploaded = new Map();
+  for (const asset of assets) {
+    // @todo need to teach ts the right type from useAssets
+    if ("meta" in asset && "family" in asset.meta) {
+      uploaded.set(asset.meta.family, { label: asset.meta.family });
+    }
+  }
+  return [...system, ...uploaded.values()];
+};
+
+const ItemMenu = ({ onDelete }: { onDelete: () => void }) => {
   return (
-    <Flex gap="3" direction="column" css={{ padding: "$1", paddingTop: "$2" }}>
+    <DropdownMenu>
+      <DropdownMenuTrigger asChild>
+        <IconButton aria-label="Font menu button">
+          <DotsHorizontalIcon />
+        </IconButton>
+      </DropdownMenuTrigger>
+      <DropdownMenuPortal>
+        <DropdownMenuContent>
+          <DropdownMenuItem
+            onSelect={() => {
+              onDelete();
+            }}
+          >
+            <Text>Delete font</Text>
+          </DropdownMenuItem>
+        </DropdownMenuContent>
+      </DropdownMenuPortal>
+    </DropdownMenu>
+  );
+};
+
+type FontsManagerProps = {
+  value: string;
+  onChange: (value: string) => void;
+};
+
+export const FontsManager = ({ value, onChange }: FontsManagerProps) => {
+  const { assets, onSubmitAssets, onActionData, onDelete } = useAssets("font");
+
+  const handleDelete = (family: string) => {
+    // One family may have multiple assets for different formats, so we need to delete them all.
+    const ids = assets
+      .filter(
+        (asset) =>
+          // @todo need to teach ts the right type from useAssets
+          "meta" in asset &&
+          "family" in asset.meta &&
+          asset.meta.family === family
+      )
+      .map((asset) => asset.id);
+    onDelete(ids);
+  };
+
+  const fontItems = useMemo(() => getItems(assets), [assets]);
+  const selectedItem =
+    fontItems.find((item) => item.label === value) ?? fontItems[0];
+
+  const { items, getComboboxProps, getMenuProps, getItemProps, getInputProps } =
+    useCombobox({
+      items: fontItems,
+      value: selectedItem,
+      itemToString: (item) => item?.label ?? "",
+      onItemSelect: (value) => {
+        if (value !== null) {
+          onChange(value.label);
+        }
+      },
+    });
+
+  const { isEmpty, ...menuProps } = getMenuProps();
+
+  return (
+    <Flex
+      gap="3"
+      direction="column"
+      css={{ padding: "$1", paddingTop: "$2", height: 460, overflow: "hidden" }}
+    >
       <Box css={{ padding: "$2" }}>
         <AssetUpload
           onSubmit={onSubmitAssets}
           onActionData={onActionData}
           type="font"
         />
       </Box>
-      {assets.map((asset) => asset.name)}
+
+      <Flex
+        {...getComboboxProps()}
+        css={{ flexDirection: "column", gap: "$3" }}
+      >
+        <TextField
+          type="search"
+          prefix={<MagnifyingGlassIcon />}
+          {...getInputProps()}
+        />
+        <fieldset>
```

Comment:
This was supposed to be an example only. In our code we should use something from DS or if it's missing, Flex with as prop?